### PR TITLE
added auto mount script

### PR DESCRIPTION
**Pull request by guruswarupa**
# Pull Request

## added auto mount script
- Automount disks script added.

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: tabs/system-setup/9-auto-mount.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        $ESCALATION_TOOL mkdir -p "$mount_point"
+    else
+        printf "%b\n" "Mount point already exists."
+    fi
+}
+
+# Function to update /etc/fstab with a comment on the first line and the actual entry on the second line
+update_fstab() {
+    printf "%b\n" "Adding entry to /etc/fstab..."
+    $ESCALATION_TOOL  cp /etc/fstab /etc/fstab.bak # Backup fstab
```
Comment (nnyyxxxx) [inline tabs/system-setup/9-auto-mount.sh:54]:
```suggestion
    $ESCALATION_TOOL cp /etc/fstab /etc/fstab.bak # Backup fstab
```

File: tabs/system-setup/9-auto-mount.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    # Prepare the comment and the fstab entry
+    comment="# Mount for /dev/$NAME"
+    fstab_entry="UUID=$UUID $mount_point $FSTYPE defaults 0 2"
+
+    # Append the comment and the entry to /etc/fstab
+    printf "%b\n" "$comment" | $ESCALATION_TOOL  tee -a /etc/fstab > /dev/null
+    printf "%b\n" "$fstab_entry" | $ESCALATION_TOOL  tee -a /etc/fstab > /dev/null
+    printf "%b\n" "" | $ESCALATION_TOOL tee -a /etc/fstab > /dev/null
```
Comment (nnyyxxxx) [inline tabs/system-setup/9-auto-mount.sh:63]:
```suggestion
    printf "%b\n" "" | "$ESCALATION_TOOL" tee -a /etc/fstab > /dev/null
```
Every $ESCALATION_TOOL needs quoted.

File: tabs/system-setup/9-auto-mount.sh [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        printf "%b\n" "Mount point doesn't exist. Creating it..."
+        $ESCALATION_TOOL mkdir -p "$mount_point"
+    else
+        printf "%b\n" "Mount point already exists."
+    fi
+}
+
+# Function to update /etc/fstab with a comment on the first line and the actual entry on the second line
+update_fstab() {
+    printf "%b\n" "Adding entry to /etc/fstab..."
```
Comment (nnyyxxxx) [inline tabs/system-setup/9-auto-mount.sh:53]:
```suggestion
    printf "%b\n" "${YELLOW}Adding entry to /etc/fstab...${RC}"
```

File: tabs/system-setup/9-auto-mount.sh [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+# Function to get UUID and FSTYPE of the selected drive
+get_uuid_fstype() {
+    UUID=$(blkid -s UUID -o value "${partition}")
+    FSTYPE=$(lsblk -no FSTYPE "${partition}")
+    NAME=$(lsblk -no NAME "${partition}")
+
+    if [ -z "$UUID" ]; then
+        printf "%b\n" "Failed to retrieve the UUID. Exiting."
```
Comment (nnyyxxxx) [inline tabs/system-setup/9-auto-mount.sh:29]:
```suggestion
        printf "%b\n" "${RED}Failed to retrieve the UUID. Exiting.${RC}"
```

File: tabs/system-setup/9-auto-mount.sh [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        exit 1
+    fi
+
+    if [ -z "$FSTYPE" ]; then
+        printf "%b\n" "Failed to retrieve the filesystem type. Exiting."
```
Comment (nnyyxxxx) [inline tabs/system-setup/9-auto-mount.sh:34]:
```suggestion
        printf "%b\n" "${RED}Failed to retrieve the filesystem type. Exiting.${RC}"
```

File: tabs/system-setup/9-auto-mount.sh [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    printf "%b\n" "Available drives and partitions:"
+    lsblk -o NAME,SIZE,FSTYPE,MOUNTPOINT,LABEL,UUID | grep -v 'loop' # list all non-loop devices
+    printf "\n"
+    printf "%b\n" "Enter the drive/partition name (e.g., sda1, sdb1): "
+    read -r drive_name
+    # Check if the input is valid
+    if lsblk | grep -q "${drive_name}"; then
+        partition="/dev/${drive_name}"
+    else
+        printf "%b\n" "Invalid drive/partition name!"
```
Comment (nnyyxxxx) [inline tabs/system-setup/9-auto-mount.sh:17]:
```suggestion
        printf "%b\n" "${RED}Invalid drive/partition name!${RC}"
```

File: tabs/system-setup/9-auto-mount.sh [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+# Function to create a mount point
+create_mount_point() {
+    printf "%b\n" "Enter the mount point path (e.g., /mnt/hdd): "
+    read -r mount_point
+    if [ ! -d "$mount_point" ]; then
+        printf "%b\n" "Mount point doesn't exist. Creating it..."
+        $ESCALATION_TOOL mkdir -p "$mount_point"
+    else
+        printf "%b\n" "Mount point already exists."
```
Comment (nnyyxxxx) [inline tabs/system-setup/9-auto-mount.sh:47]:
```suggestion
        printf "%b\n" "${RED}Mount point already exists.${RC}"
```

File: tabs/system-setup/9-auto-mount.sh [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+        exit 1
+    fi
+}
+
+# Function to create a mount point
+create_mount_point() {
+    printf "%b\n" "Enter the mount point path (e.g., /mnt/hdd): "
+    read -r mount_point
+    if [ ! -d "$mount_point" ]; then
+        printf "%b\n" "Mount point doesn't exist. Creating it..."
```
Comment (nnyyxxxx) [inline tabs/system-setup/9-auto-mount.sh:44]:
```suggestion
        printf "%b\n" "${YELLOW}Mount point doesn't exist. Creating it..${RC}."
```

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**